robot/impl/local_robot.go

```diff
@@ -1204,6 +1204,30 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 		return
 	}
 
+	r.logger.CInfo(ctx, "running module setup phase")
+	// Diff configs before resolving dependencies to find what modules have 
+	// been added or updated. We run the setup phase for these modules before 
+	// proceeding with reconfiguration.
+	diff, err := config.DiffConfigs(*r.Config(), *newConfig, r.revealSensitiveConfigDiffs)
```

Comment:
I admit it's a little ugly to `DiffConfigs` twice - once here and again [later on after we resolve deps](https://github.com/viamrobotics/rdk/pull/4457/files#diff-0b975da6011656cd3f2caf0435023573bdb5ac7c6a425e7917dd5cc9d6be63e4R1286). However, we do need to this diff earlier to get Added + Modified modules before we cache the config, so I think diffing twice is the simplest way to accomplish that.

Open to suggestions on alternatives. The one other thing I considered was delaying the cache step, but that felt like a bigger/riskier change.

Reply:
we actually don't need to diff - we can just run the setup phase for all modules in the new config